tools/eslint-rules/new-with-error.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @fileoverview Require `throw new Error()` rather than `throw Error()`
+ * @author Rich Trott
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+  return {
+    'ThrowStatement': function(node) {
+      const arg = node.argument;
+      if ((arg.type === 'CallExpression') && (arg.callee.name === 'Error')) {
```

Comment:
`/^[A-Z\w]*Error^/.test(arg.callee.name)` should catch all other error types, I think.